prime-router/cleanslate.sh

```diff
@@ -311,7 +318,10 @@ function populate_vault() {
 function cleanup() {
   info "> Cleaning up your environment..."
   docker_decompose
-  take_directory_ownership
+  if [[ ${RETAKE_OWNERSHIP?} != 0 ]]; then
+    info "Change directory ownership" 
+    take_directory_ownership
```

Comment:
I would do this in `take_directory_ownership` itself instead of the call-sites:

```bash

# Takes ownership of some directories that are potentially shared with container instances
# Parameter: TAKE_OWNERSHIP value
function take_directory_ownership() {
  if [[ ${1} != 0 ]]; then
    info "Taking ownership of directories (may require elevation)..."
    TARGETS=(
      "./build/"
      "./docs/"
      "./.gradle/"
      "./.vault/"
    )

    # If you have no sudo, then this is a no-op and we warn
    SUDO_CMD=$(which sudo | head -n 1)
    if [[ -z "${SUDO_CMD}" ]]; then
      warning "You do not apear to have 'sudo'; it is highly probable that incorrect file permissions will lead to sftp upload failures when you run tests"
    fi

    for d in ${TARGETS[*]}; do
      echo -ne "    - ${d?}..."
      if [[ -d "${d?}" ]]; then
        # No '?' in the variable reference, it's fine for it to not exist, this may just be a no-op
        # and we warned you about it just above
        ${SUDO_CMD} chown -R "$(id -u -n):$(id -g -n)" "${d?}"
        ${SUDO_CMD} chmod -R a+w "${d?}"
        echo "DONE"
      else
        echo "ABSENT(OK)"
      fi
    done
  else
    info "Leaving directories alone..."
  fi
}
```

and when you call it, you do:
`take_directory_ownership ${RETAKE_OWNERSHIP?}` (or whatever you end up calling the variable)

Reply:
Change made

Reply:
Shouldn't it be defaulted as 0?

Reply:
Shouldn't it be defaulted to 0 and not 1?